apps/web/vibes/soul/examples/sections/cart/electric.tsx

```diff
@@ -1,24 +1,247 @@
-      lineItemAction={lineItemAction}
-    />
+    <div className="p-4">
+      <Form action={resetCart}>
+        <Button type="submit" variant={'ghost'}>
+          Reset cart
+        </Button>
+      </Form>
+      <Cart
+        cart={await getCart()}
+        checkoutAction={checkoutAction}
+        key={await getCartId()}
+        lineItemAction={lineItemAction}
+      />
+    </div>
   );
 }
+
+const defaultLineItems: CartLineItem[] = [
+  {
+    id: '1',
+    title: 'Philodendron Imperial Red',
+    subtitle: 'Indoor Plant',
+    price: '$46',
+    image: {
+      src: 'https://rstr.in/monogram/vibes/-kv08IvX08j',
+      alt: 'Philodendron Imperial Red',
+    },
+    quantity: 1,
+  },
+  {
+    id: '2',
+    title: 'Caladium',
+    subtitle: 'Indoor / Outdoor Plant',
+    price: '$24',
+    image: {
+      src: 'https://rstr.in/monogram/vibes/AaZW4j2VTd4',
+      alt: 'Caladium',
+    },
+    quantity: 2,
+  },
+];
+
+export async function checkoutAction(
+  prevState: Awaited<SubmissionResult | null>,
+): Promise<SubmissionResult | null> {
+  'use server';
+
+  // Simulate a network request
+  await new Promise((resolve) => setTimeout(resolve, 1000));
+
+  return prevState;
+}
+
+export async function lineItemAction(
+  prevState: Awaited<{
+    lineItems: CartLineItem[];
+    lastResult: SubmissionResult | null;
+  }>,
+  formData: FormData,
+): Promise<{
+  lineItems: CartLineItem[];
+  lastResult: SubmissionResult | null;
+}> {
+  'use server';
+
+  const submission = parseWithZod(formData, { schema: cartLineItemActionFormDataSchema });
+
+  if (submission.status !== 'success') {
+    return {
+      ...prevState,
+      lastResult: submission.reply({ formErrors: ['Boom!'] }),
+    };
+  }
+
+  switch (submission.value.intent) {
+    case 'increment': {
+      await incrementLineItem(submission.value.id);
+
+      break;
+    }
+    case 'decrement': {
+      await decrementLineItem(submission.value.id);
+
+      break;
+    }
+    case 'delete': {
+      await deleteLineItem(submission.value.id);
+
+      break;
+    }
+
+    default: {
+      return prevState;
+    }
+  }
+
+  return {
+    lineItems: (await getCart()).lineItems,
+    lastResult: submission.reply({ resetForm: true }),
+  };
+}
+
+const cartSchema = z.object({
+  id: z.string(),
+  lineItems: z.array(
+    z.object({
+      id: z.string(),
+      title: z.string(),
+      subtitle: z.string(),
+      price: z.string(),
+      quantity: z.number(),
+      image: z.object({
+        src: z.string(),
+        alt: z.string(),
+      }),
+    }),
+  ),
+  summaryItems: z.array(
+    z.object({
+      label: z.string(),
+      value: z.string(),
+    }),
+  ),
+  total: z.string(),
+});
+
+type SummaryItems = Array<{ label: string; value: string }>;
+interface Cart<CartLineItem> {
+  id: string;
+  lineItems: CartLineItem[];
+  summaryItems: SummaryItems;
+  total: string;
+}
+
+/**
+ * Replace these with your own implementation
+ */
```

Comment:
This JSDoc comment is going to be associated with `getCart` but it seems to be intended for the rest of the file, not just this function. I'd suggest not using JSDoc and instead using a regular multi-line comment

Reply:
Went ahead and removed it since, I think it's obvious they'll need their own implementation